addon/components/hot-replacement-component.js

```diff
@@ -61,8 +62,10 @@ const HotReplacementComponent = Ember.Component.extend(HotComponentMixin, {
   }).volatile(),
 
   __willLiveReload (event) {
-    if (matchingComponent(this.get('baseComponentName'), event.modulePath)) {
+    const baseComponentName = this.get('baseComponentName');
+    if (matchingComponent(baseComponentName, event.modulePath)) {
       event.cancel = true;
+      clearRequirejs(this, baseComponentName);
```

Comment:
Oh, I see the difference now. The name is confusing. I will address that on a separate PR. `willHotReload` is happening *after* we load the file. I was expecting it to happen before it loads the file.